api/storage/client.go

```diff
@@ -49,6 +50,30 @@ func (c *Client) Show(tags []names.StorageTag) ([]params.StorageInstance, error)
 	return all, allErr.Combine()
 }
 
+// PoolList lists pools according to a given filter.
+// If no filter was provided, this will return a list
+// of all pools.
+func (c *Client) PoolList(types, names []string) ([]params.PoolInstance, error) {
+	args := params.PoolListFilter{
```

Comment:
Maybe call this type "StoragePoolFilter", so it can be reused for other pool-related API calls that need to do filtering.